ui/ceval/src/view.ts

```diff
@@ -298,10 +313,23 @@ export function renderPvs(ctrl: ParentCtrl): VNode | undefined {
             instance.setHovering(getElFen(el), getElUci(e));
             const pvBoard = (e.target as HTMLElement).getAttribute('data-board');
             if (pvBoard) {
+              pvMoves = getElPvMoves(e);
+              pvIndex = pvMoves.indexOf(pvBoard);
```

Comment:
This won't work when a position repeats, which I'm pretty sure can happen in Stockfish lines.

Reply:
I did not test with a specific example but I think it works because ```pvMoves = getElPvMoves(e);``` fetches the moves from the line that is hovered.  ```pvMoves.indexOf(pvBoard);``` gets me the index of the current move so I can go back and forth inside that line.

Reply:
Yeah I get what you are saying, I have to look into this with a specific example.

Reply:
I think my last commit fixes the issue. It ensures we get the correct move index.